src/components/UpdatePreferences.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import {
+  updateUserPreferences,
+  getUserPreferences
+} from "../store/reducers/userPreferencesReducer";
+
+class UpdatePreferences extends Component {
+  constructor() {
+    super();
+    this.updates = [
+      { activityId: 0 },
+      { activityId: 1 },
+      { activityId: 2 },
+      { activityId: 3 },
+      { activityId: 4 }
+    ];
+  }
+  handleCheck = evt => {
+    this.updates.map(updateObj => {
+      if (updateObj.activityId === Number(evt.target.id)) {
+        evt.target.checked
+          ? (updateObj["include"] = true)
+          : (updateObj["include"] = false);
+      }
+    });
+  };
+
+  handleChange = evt => {
+    this.updates.map(updateObj => {
+      if (updateObj.activityId === Number(evt.target.id))
+        updateObj[evt.target.name] =
+          evt.target.name === "duration"
+            ? Number(evt.target.value)
```

Comment:
These two different event.target.names result in the same 
`Number(evt.target.value)`